src/Microsoft.Health.Dicom.Core/Features/Workitem/Model/WorkitemInstanceIdentifier.cs

```diff
@@ -30,6 +32,8 @@ public class WorkitemInstanceIdentifier
 
         public string WorkitemUid { get; }
 
+        public long? Watermark { get; }
```

Comment:
WorkitemUid is the one that makes it unique in the system. Watermark is used only to avoid two parties updating the same workitem with the outdated information. We could include Watermark in the ToString, but may avoid using it in the GetHashCode calculation.